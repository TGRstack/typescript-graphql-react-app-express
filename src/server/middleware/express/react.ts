import paths from '!server/config/paths'
import logger from '!server/modules/logger'
import * as express from 'express'
import * as fs from 'fs'
import * as path from 'path'

export default function expressReact(app: express.Application) {
  if (fs.existsSync(paths.client)) {
    // tslint:disable-next-line no-console
    logger.info(`[Loaded] React Client: ${path.relative('.', paths.client)}`)

    app.use(express.static(paths.app))
    app.get('*', ({}, res: Express.Response) => {
      res.sendFile(paths.client)
    })
  } else {
    logger.error(`No React Client "index.html" at: ${paths.client}`)
  }
}
