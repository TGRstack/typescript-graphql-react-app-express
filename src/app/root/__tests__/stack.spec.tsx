import { shallow } from 'enzyme'
import * as React from 'react'
import * as ReactDOM from 'react-dom'

import App from '../stack'

describe('Application Integration', () => {
  it('loads w/o crashing', () => {
    const root = document.createElement('div')
    root.setAttribute('id', 'app')
    ReactDOM.render(<App />, root)
    // ReactDOM.createRoot(root).render(<App />)
  })
  it('matches the snapshot', () => {
    const wrapper = shallow(<App />)
    expect(wrapper).toMatchSnapshot()
  })
})
