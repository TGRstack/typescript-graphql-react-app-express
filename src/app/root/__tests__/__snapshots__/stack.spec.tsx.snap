// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Application Integration matches the snapshot 1`] = `ShallowWrapper {}`;
